pkg/snapstore/gcs_snapstore.go

```diff
@@ -74,7 +87,7 @@ func NewGCSSnapStoreFromClient(bucket, prefix, tempDir string, maxParallelChunkU
 
 // Fetch should open reader for the snapshot file from store.
 func (s *GCSSnapStore) Fetch(snap brtypes.Snapshot) (io.ReadCloser, error) {
-	objectName := path.Join(snap.Prefix, snap.SnapDir, snap.SnapName)
+	objectName := path.Join(s.getPrefix(snap), snap.SnapDir, snap.SnapName)
```

Comment:
@abdasgupta Can you please check if this change is required for the changes for folder structure change also?

Reply:
@kris94 Would you please explain that why do we need it here? Like under what situation we would have empty prefix in snapshot but a path in GCS Store?  Elsewhere,`List` function is actually making sure that every snapshot should have prefix field set.

Reply:
The rationale behind the question is, if there is a bug we would like to include a fix in the release for backup compaction that we are planning to cut this week.